### PR TITLE
Change string concation to template literals

### PR DESCRIPTION
**Pull request by mtrivera**
Changes proposed in this pull request:
- Replace all string concatenations with template literals in `/dkim.js`.
- updates #2129

### CODE REVIEW

File: dkim.js [hunk 1]
```diff
@@ -173,12 +173,12 @@ class DKIMObject {
...
         this.debug([
-            this.identity + ':',
-            'a=' + this.fields.a,
-            'c=' + this.headercanon + '/' + this.bodycanon,
-            'h=' + this.signed_headers,
+            `${this.identity}:`,
+            `a=${this.fields.a}`,
+            `c=${this.headercanon}/${this.bodycanon}`,
+            `h=${this.signed_headers}`,
         ].join(' '));
```
Comment (msimerson) [inline dkim.js]: consider making this a single template string, instead of 4 array elements joined with a space.

File: dkim.js [hunk 2]
```diff
@@ -241,17 +241,17 @@ class DKIMObject {
         if (this.run_cb) return;
 
         const bh = this.bh.digest('base64');
-        this.debug(this.identity + ':' +
-            ' bodyhash=' + this.fields.bh +
-            ' computed=' + bh);
+        this.debug(`${this.identity}:
+             bodyhash=${this.fields.bh}
+             computed=${bh}`);
```
Comment (msimerson) [inline dkim.js]: in template literals, all that leading whitespace is now part of the string. Oops.

File: dkim.js [hunk 3]
```diff
@@ -336,9 +336,9 @@ class DKIMObject {
                         s = s.replace(/[-[\]{}()*+?.,\\^$|#\s]/g, "\\$&");
                         // Make * a non-greedy match against anything except @
                         s = s.replace('\\*','[^@]*?');
-                        const reg = new RegExp('^' + s + '@');
-                        self.debug(self.identity + ': matching ' + self.dns_fields.g +
-                            ' against i=' + self.fields.i + ' regexp=' + reg.toString());
+                        const reg = new RegExp(`^${s}@`);
+                        self.debug(`${self.identity}: matching ${self.dns_fields.g}
+                             against i=${self.fields.i} regexp=${reg.toString()}`);
```
Comment (msimerson) [inline dkim.js]: same as above, extra whitespace.

**Comment by mtrivera** [comment]
Made corrections based on feedback

**Pull request merged**